xwiki-commons-core/xwiki-commons-extension/xwiki-commons-extension-handlers/xwiki-commons-extension-handler-jar/src/main/java/org/xwiki/extension/jar/internal/handler/JarExtensionJobFinishingListener.java

```diff
@@ -149,7 +150,7 @@ private void popUninstallLevel()
         ExecutionContext context = this.execution.getContext();
 
         if (context != null) {
-            Stack<UninstalledExtensionCollection> extensions = getUninstalledExtensionCollectionStack(false);
+            Deque<UninstalledExtensionCollection> extensions = (Deque<UninstalledExtensionCollection>) getUninstalledExtensionCollectionStack(false);
```

Comment:
That's not what Sonar suggest to do. The implementation is still a Stack here, you just cast it.